packages/siwx/src/core/SIWXMessenger.ts

```diff
@@ -47,6 +80,13 @@ export abstract class SIWXMessenger {
     return Object.assign(params, methods)
   }
 
+  /**
+   * This method generates the expiration time based on the `notBefore` value and the `expiration` value.
+   * You may override this method to provide a custom expiration time.
+   *
+   * @param Pick<SIWXMessage.Input, 'notBefore'>
+   * @returns string | undefined - The expiration time in `stringifyDate` format or undefined if not available
+   */
```

Comment:
is this a utility? or is this core to the package?

Reply:
this is an abstract class that must be extended to provide the functionality, this is only predefined logic